### PR TITLE
Reject unnecessary messages - Closes #4424

### LINKED ISSUE #4424: Reject unnecessary messages and ban a peer
### Expected behavior

A node should not send internal messages to a lisk node as it's not needed which creates an extra overhead.

### Actual behavior

Its accepting messages that are internal and not related to `lisk-sdk` and should be rejected and a peer should be banned as its not suppose to send these messages.

### Which version(s) does this affect? (Environment, OS, etc...)
`3.0`

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

Peers can send unnecessary or internal messages that are not part of the protocol.

### How did I solve it?

Rejecting the message and banning the peer.

### How to manually test it?

Send messages like `INVALID` or something and see if the receiving node rejects the connection

### Review checklist

- [x] The PR resolves #4424
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)

### CODE REVIEW

File: elements/lisk-p2p/src/constants.ts [hunk 1]
```diff
@@ -55,6 +55,8 @@ export const DEFAULT_PRODUCTIVITY = {
 // Peer inbound constants
 export const DEFAULT_PING_INTERVAL_MAX = 60000;
 export const DEFAULT_PING_INTERVAL_MIN = 20000;
+export const SOCKET_PING_MESSAGE = '#1';
```
Comment (diego-G) [inline elements/lisk-p2p/src/constants.ts]: Where is this coming from? It looks too specific. I would suggest to have a regEx capturing similar cases.

File: elements/lisk-p2p/src/constants.ts [hunk 2]
```diff
@@ -66,6 +68,9 @@ export const DEFAULT_MAX_RECONNECT_TRIES = 3;
 
 // Disconnect statuses
 export const INTENTIONAL_DISCONNECT_CODE = 1000;
+export const INVALID_MESSAGE_RECEIVED_CODE = 4009;
```
Comment (diego-G) [inline elements/lisk-p2p/src/constants.ts]: Why 4009 instead of 4109 or 4409?

File: elements/lisk-p2p/src/peer/base.ts [hunk 3]
```diff
@@ -498,6 +525,7 @@ export class Peer extends EventEmitter {
 		this._wsMessageRate =
 			(this._wsMessageCount * RATE_NORMALIZATION_FACTOR) / this._rateInterval;
 		this._wsMessageCount = 0;
+		this._badMessageCount = 0;
```
Comment (diego-G) [inline elements/lisk-p2p/src/peer/base.ts]: I suggest to use a more meaningful naming. What is a bad message?

File: elements/lisk-p2p/src/p2p.ts [hunk 4]
```diff
@@ -640,6 +640,20 @@ export class P2P extends EventEmitter {
 	}
 
 	private async _startPeerServer(): Promise<void> {
+		this._scServer.on(
+			'handshake',
```
Comment (diego-G) [inline elements/lisk-p2p/src/p2p.ts]: Shall we move here the blacklist check present in `connection` event?

File: elements/lisk-p2p/test/integration/reject_unwanted_events.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		for (const p2p of p2pNodeList) {
+			if (p2p.nodeInfo.wsPort === connectedPeerOfNode2.wsPort) {
+				const peerOfNode1 = firstNode['_peerPool']['_peerMap'].get(
+					`${SEED_PEER_IP}:${secondNode.nodeInfo.wsPort}`,
+				);
+				if (peerOfNode1 && peerOfNode1['_socket']) {
+					for (let num = 0; num < 50; num++) {
+						peerOfNode1['_socket'].send(INVALID_MESSAGE_1, {});
```
Comment (diego-G) [inline elements/lisk-p2p/test/integration/reject_unwanted_events.ts]: We only reject that specific message '#1'. The PR states different `Send messages like INVALID or something and see if the receiving node rejects the connection`

File: elements/lisk-p2p/src/constants.ts (hunk 2 above)
Comment (ishantiw) [reply elements/lisk-p2p/src/constants.ts]: Its a socketcluster specific close code on invalid raw messages,
https://github.com/SocketCluster/socketcluster-server/blob/master/scserversocket.js#L106

File: elements/lisk-p2p/src/constants.ts [hunk 6]
```diff
@@ -66,6 +68,9 @@ export const DEFAULT_MAX_RECONNECT_TRIES = 3;
...
+export const INVALID_MESSAGE_RECEIVED_REASON =
+	'Peer should not send bad or internal messages';
```
Comment (diego-G) [inline elements/lisk-p2p/src/constants.ts]:
```suggestion
	'Peer should not send invalid or internal messages';
```

File: elements/lisk-p2p/src/p2p.ts (hunk 4 above)
Comment (diego-G) [reply elements/lisk-p2p/src/p2p.ts]: Isn't it redundant now? My suggestion was to move the code...

**Pull request merged**